include/xrpl/basics/BasicConfig.h

```diff
@@ -215,7 +218,7 @@ class Section
 class BasicConfig
 {
 private:
-    std::map<std::string, Section, boost::beast::iless> map_;
+    std::unordered_map<std::string, Section> map_;
```

Comment:
What does the conversion from `map` to `unordered_map` get us? And why did you remove the case-insensitive comparison?

Reply:
Conversion to unordered map  came  from the usage of the config - it initialized once and then only used for querying. Which is generally more effective with unordered map. 

`boost::beast::iless`  seems like error or not finished refactoring here. 
`IniFileSections`, initializer of the   `map_`,  is not case-insensitive.
`lookup_`, values in the sections, is  not case-insensitive.